src/emission_profile.jl

```diff
@@ -0,0 +1,82 @@
+#imports
+using Plots
+
+#functions
+"""
+From Reynolds (2020) (Eq2-4)
+Calculates the radius of the inner most stable circular orbit of the a black hole,
+with spin a_star, and mass M.
+"""
+function r_isco(a_star, M)
+    
+    a = a_star
+    r_g = 2*G*M/(c^2)
+    z1 = 1+∛(1-a^2)*(∛(1+a)+∛(1-a))
+    z2 = √(3*a^2+z1^2)
+    if a >= 0
+        r_isco = (3+z2-(3-z1)*√(3+z1+2*z2))*r_g
+    elseif a < 0
+        r_isco = (3+z2+(3-z1)*√(3+z1+2*z2))*r_g
+    end
+end
+
+"""
+From Page & Thorne (1974) (Eq15n)
+Calculates the function f at radius r, for a black hole of spin a_star, and mass M, 
+which is used to calculate the flux given by its accretion disk.
+"""
+function f(r, a_star, M)
+
+    R_isco = r_isco(a_star, M)
+  
+    x = √(r/M)
+    x0 = √(R_isco/M)
+    x1 = 2*cos((1/3)*(acos(a_star))-(π/3))
+    x2 = 2*cos((1/3)*(acos(a_star))+(π/3))
+    x3 = -2*cos((1/3)*(acos(a_star)))
+
+    f = (3/(2*M))*(1/(x^2*(x^3-(3*x)+(2*a_star))))*(x-x0-((3/2)*a_star*log(x/x0)) 
+            - (3*(x1-a_star)^2)/(x1*(x1-x2)*(x1-x3))*log((x-x1)/(x0-x1)) 
+            - (3*(x2-a_star)^2)/(x2*(x2-x1)*(x2-x3))*log((x-x2)/(x0-x2)) 
+            - (3*(x3-a_star)^2)/(x3*(x3-x1)*(x3-x2))*log((x-x3)/(x0-x3)))
+end
+
+function diss(mdot, r, a_star, M)
+    diss = mdot*f(r, a_star, M)/(4*π*r)
+end
+
+function mdot(M)
+    L_edd = 3e4*L_☼*(M/M_☼)
+    Mdot = -L_edd/(c^2*η)
+    mdot = 0.1*Mdot
+end
+
+function temp(r, a_star, M)
+    m_dot = mdot(M)
+    temp = (diss(m_dot, r, a_star, M)/σ_SB)^(1/4)
+end
+
+"""
+From Fanton et al. (1997) (Eq78)
+"""
+function temp_obs(r, a_star, M, g)
+    temp = temp(r, a_star, M)
+    temp_obs =  g*temp
+end
+
+# constants
+G = 6.67e-11
+c = 3e8
+L_☼ = 3.8e26
+M_☼ = 1.99e30
+σ_SB = 5.67e-8
+
+η = 0.1
+# M = 10*M_☼
+# a_star = 0.1
+
+module EmissionProfile
+
+export temp_obs, r_isco
+
+end
```

Comment:
Replace this with just
```julia
export temp_obs, r_isco
```
and then we'll add a line to `src/AccretionFormulae.jl` which `include`s this file -- then it's in the scope of the AccretionFormulae module, and will be exported from there.

Reply:
Ok looks good, should I add the `include` line to `src/AccretionFormulae.jl` now, or is that something you'll need to do?

Reply:
I experimented with this in the copy in my DevEnv, and got the following error:

```julia
ERROR: LoadError: importing AccretionFormulae into Main conflicts with an existing identifier
Stacktrace:
  [1] eval
    @ .\boot.jl:360 [inlined]
  [2] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
    @ Base .\loading.jl:1116
  [3] invokelatest(::Any, ::Any, ::Vararg{Any, N} where N; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ Base .\essentials.jl:708
  [4] invokelatest(::Any, ::Any, ::Vararg{Any, N} where N)
    @ Base .\essentials.jl:706
  [5] inlineeval(m::Module, code::String, code_line::Int64, code_column::Int64, file::String; softscope::Bool)
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\eval.jl:211
  [6] (::VSCodeServer.var"#58#62"{Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\eval.jl:155
  [7] withpath(f::VSCodeServer.var"#58#62"{Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams}, path::String)
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\repl.jl:185
  [8] (::VSCodeServer.var"#57#61"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\eval.jl:153
  [9] hideprompt(f::VSCodeServer.var"#57#61"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\repl.jl:36
 [10] (::VSCodeServer.var"#56#60"{Bool, Bool, Bool, Module, String, Int64, Int64, String, VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\eval.jl:124
 [11] with_logstate(f::Function, logstate::Any)
    @ Base.CoreLogging .\logging.jl:491
 [12] with_logger
    @ .\logging.jl:603 [inlined]
 [13] (::VSCodeServer.var"#55#59"{VSCodeServer.ReplRunCodeRequestParams})()
    @ VSCodeServer c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\eval.jl:201
 [14] #invokelatest#2
    @ .\essentials.jl:708 [inlined]
 [15] invokelatest(::Any)
    @ Base .\essentials.jl:706
 [16] macro expansion
    @ c:\Users\Dan\.vscode\extensions\julialang.language-julia-1.5.11\scripts\packages\VSCodeServer\src\eval.jl:34 [inlined]
 [17] (::VSCodeServer.var"#53#54")()
    @ VSCodeServer .\task.jl:411
in expression starting at c:\Users\Dan\Documents\Uni\Year 4\Project\GitHub\DevEnv\dev\AccretionFormulae\src\test.jl:4
```

Is there something that I need to clear or import again to fix this?

For reference `AccretionFormulae.jl` looks like this at the moment:
```julia
module AccretionFormulae

include("emission_profile.jl")

end # module
```

and `emission_profile.jl` doesn't seem to give any errors when run.

Reply:
That error looks like you have to just restart you Julia environment, but if that doesn't fix it, just push your code tomorrow, and I'll see what the problem is!